pkg/enqueue/Client/Driver/RabbitMqDriver.php

```diff
@@ -16,16 +16,14 @@ protected function doCreateQueue(string $transportQueueName): InteropQueue
         $queue = parent::doCreateQueue($transportQueueName);
 
         $driverOptions = $config->getDriverOptions();
-        $islazyQueue = false;
-        $lazyQueueArray = array_filter($driverOptions, function ($arr) {
-            return array_key_exists('rabbit_mq_lazy_queues', $arr);
-        });
+        $lazyQueueArray = $driverOptions['rabbit_mq_lazy_queues'] ?? [];
+        $isLazyQueue = in_array($transportQueueName, $lazyQueueArray, true);
 
-        $islazyQueue = array_key_exists($transportQueueName, $lazyQueueArray[0]['rabbit_mq_lazy_queues']);
-
-        $queue->setArguments(['x-max-priority' => 4]);
-        if ($islazyQueue) {
-            $queue->setArguments(['x-queue-mode' => 'lazy']);
+        if (true == $isLazyQueue) {
+            $queue->setArguments(
+                ['x-queue-mode' => 'lazy', 'x-max-priority' => 4]);
```

Comment:
Use `$queue->setArgument(key, value)` instead. That way you can drop the inconvenient `if...else`.

```php
$queue->setArguments(['x-max-priority' => 4]);
if ($isLazyQueue) {
        $queue->setArgument('x-queue-mode' => 'lazy');
}
```
This ensures that adding new options is kept simple and does not influence too many lines in version control (not to mention creating a devilish `if` :smiley: )

`true == $isLazyQueue` is not really needed. PHP will cast `$isLazyQueue` to boolean value anyway, and since there is no type check there is no point to write `true ==`.

Lastly, even if you'd like to stick to `setArguments` method, options array should be spread across lines. Like so:
```php
$queue->setArguments([
    'x-queue-mode' => 'lazy', 
    'x-max-priority' => 4,
]);
```
Reason being, if anyone decides to add another option, it will appear as a new line and will not cause conflicts between different PRs. This is also the reason that last array element has a comma at the end.

Reply:
Cool, thanks for the pointers @Steveb-p  - PHP is definitely not my primary language, so these are really helpful.